_posts/en/newsletters/2025-01-31-newsletter.md

```diff
@@ -0,0 +1,263 @@
+
+## Notable code and documentation changes
+
+_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
+[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
+repo], and [BINANAs][binana repo]._
+
+- [Bitcoin Core #31376][] extends a check that prevents miners from creating
+  block templates that exploit the [timewarp][topic time warp] bug to apply to
+  all networks, not just [testnet4][topic testnet]. This change is in
+  preparation for a possible future soft fork that would permanently fix the
+  timewarp bug.
+
+- [Bitcoin Core #31583][] updates the `getmininginfo`, `getblock`,
+  `getblockheader`, `getblockchaininfo` and `getchainstates` RPC commands to now
+  return a `nBits` field (the compact representation of the block difficulty
+  target) and a `target` field. In addition, `getmininginfo` adds a `next`
+  object that specifies the height, `nBits`, difficulty, and target for the next
+  block. To derive and obtain the target, this PR introduces the
+  `DeriveTarget()` and the `GetTarget()` helper functions. These changes are
+  useful for the implementation of [Stratum V2][topic pooled mining].
+
+- [Bitcoin Core #31590][] refactors the `GetPrivKey()` method to check pubkeys
+  for both parity bit values when retrieving private keys for an x-only pubkey
+  in a [descriptor][topic descriptors]. Previously, if the stored pubkey didn’t
+  have the correct parity bit, the private key couldn’t be retrieved and
+  transactions couldn’t be signed.
+
+- [Eclair #2982][] introduces the `lock-utxos-during-funding` configuration
+  setting, allowing [liquidity advertisement][topic liquidity advertisements]
+  sellers to mitigate liquidity griefing attacks (see Newsletter [#281][news281
+  griefing]) that could lock their UTXOs for extended periods. The default
+  setting is true, meaning that UTXOs are locked during the funding process and
+  are vulnerable. If set to false, UTXO locking is disabled and the attack can
+  be completely prevented, but this may have a negative impact on honest peers.
+  This PR also adds a configurable timeout mechanism that automatically aborts
+  incoming channels if a peer becomes unresponsive.
+
+- [BDK #1614][] adds support for using [compact block filters][topic compact
+  block filters] as specified in [BIP158][] for downloading confirmed
+  transactions. This is done by adding a BIP158 module to the `bdk_bitcoind_rpc`
+  crate, along with a new `FilterIter` type that can be used to retrieve blocks
+  from a full node that contain transactions relevant to a list of script
+  pubkeys.
```

Comment:
```suggestion
  containing transactions relevant to a list of script
  pubkeys from a full node.
```

Reply:
Dropped "full node" clause entirely.  It's possible, at least in theory, to get blocks from non-full nodes.